pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -268,19 +266,20 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 	ctx := context.TODO()
 	client := getClient(c.Token)
 
-	// TODO(p0lyn0mial): an ssh-key should be created only if the CoreOS was requested
-	// TODO(p0lyn0mial): an ssh-key must be temporal - deleted after an instance has been created.
-	if p.privateKey == nil {
-		var err error
-		p.privateKey, err = machinessh.NewPrivateKey("machine-controller")
-		if err != nil {
-			return nil, err
-		}
+	tmpSSHKeyPair, err := machinessh.NewPrivateKey("machine-controller-do-tmp-key")
```

Comment:
we could also drag the controller name all the way from main.go and use it here

Reply:
Better generate a new random name. Otherwise we run into concurrency issues here

Reply:
good point, before we had only once instance of that key. I think `UUID` is fine.